atom/common/node_bindings.cc

```diff
@@ -213,19 +229,42 @@ void NodeBindings::Initialize() {
   // Explicitly register electron's builtin modules.
   RegisterBuiltinModules();
 
-  // Init node.
-  // (we assume node::Init would not modify the parameters under embedded mode).
-  // NOTE: If you change this line, please ping @codebytere or @MarshallOfSound
+  // pass non-null program name to argv so it doesn't crash
+  // trying to index into a nullptr
   int argc = 0;
   int exec_argc = 0;
-  const char** argv = nullptr;
+  const char* prog_name = "electron";
+  const char** argv = {&prog_name};
   const char** exec_argv = nullptr;
+
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  if (env->HasVar("NODE_OPTIONS")) {
+    std::vector<std::string> disallowed = {
+        "--inspect", "--inspect-brk",    "--inspect-port", "--require",
```

Comment:
if i missed any potentially dangerous options, please let me know!

Reply:
This is a very brittle way of preventing these options from being passed. If node updates the option list and changes or adds any options, we'll have no way of knowing if something has changed that will break this list. This is advisory at best.

Reply:
What if we dissallowed the use of NODE_OPTIONS all together when running in a packaged app?

Reply:
i've addressed to best of ability, let me know what yall think?

Reply:
> What if we dissallowed the use of NODE_OPTIONS all together when running in a packaged app?

Agreed, this would solve my concern and also simplify the current solution. These options are developer focused, they don't make sense in the context of packaged apps.